js/builtins/weakrefs/holdings-multiple-values.optional.any.js

```diff
@@ -0,0 +1,69 @@
+// META script=resources/maybe-garbage-collect.js
+// ├──> maybeGarbageCollectAndCleanupAsync
+// └──> resolveGarbageCollection
+/*---
+esid: sec-properties-of-the-finalization-registry-constructor
+info: |
+  FinalizationRegistry.prototype.cleanupSome ( [ callback ] )
+
+  1. Let finalizationRegistry be the this value.
+  ...
+  5. Perform ! CleanupFinalizationRegistry(finalizationRegistry, callback).
+  ...
+
+  CleanupFinalizationRegistry ( finalizationRegistry [ , callback ] )
+
+  ...
+  3. While finalizationRegistry.[[Cells]] contains a Record cell such that cell.[[WeakRefTarget]] is ~empty~, then an implementation may perform the following steps,
+    a. Choose any such cell.
+    b. Remove cell from finalizationRegistry.[[Cells]].
+    c. Perform ? Call(callback, undefined, << cell.[[HeldValue]] >>).
+
+---*/
+
+function check(value, expectedName) {
+  let holdings = [];
+  let called = 0;
+  let finalizationRegistry = new FinalizationRegistry(function() {});
+
+  function callback(holding) {
+    called += 1;
+    holdings.push(holding);
+  }
+
+  // This is internal to avoid conflicts
+  function emptyCells(value) {
+    let target = {};
+    finalizationRegistry.register(target, value);
+
+    let prom = maybeGarbageCollectAndCleanupAsync(target);
+    target = null;
+
+    return prom;
+  }
+
+  return emptyCells(value).then(function() {
+    finalizationRegistry.cleanupSome(callback);
+    assert_equals(called, 1, expectedName);
+    assert_equals(holdings.length, 1, expectedName);
+    assert_equals(holdings[0], value, expectedName);
+  });
+}
+
+promise_test(() => {
+  assert_implements(
+    typeof FinalizationRegistry.prototype.cleanupSome === 'function',
+    'FinalizationRegistry.prototype.cleanupSome is not implemented.'
+  );
+
+  return Promise.all([
+    check(undefined, 'undefined'),
+    check(null, 'null'),
```

Comment:
Each call to "check" is actually a distinct test. In WPT, a test file is composed of one or more subtests--you can call `promise_test` more than once. We should do that here for two reasons:

- it will give implementers a clearer picture of what's going wrong when the test fails
- it will promote stability (as written, the test will fail with a message that describes whatever "check" happened to fail first. I have a feeling that won't be deterministic)

Sorry for the late-stage feedback; this only just dawned on me.

Reply:
No problem, this is good. I'm making this change now